crates/db_views/src/custom_emoji_view.rs

```diff
@@ -0,0 +1,153 @@
+use crate::structs::{CustomEmojiView, CustomEmojiViewForm};
+use diesel::{insert_into, result::Error, BelongingToDsl, ExpressionMethods, GroupedBy, QueryDsl};
+use diesel_async::RunQueryDsl;
+use lemmy_db_schema::{
+  newtypes::LocalSiteId,
+  schema::{
+    custom_emoji::{
+      dsl::{custom_emoji, local_site_id},
+      id,
+    },
+    custom_emoji_keyword::{custom_emoji_id, dsl::custom_emoji_keyword},
+  },
+  source::{
+    custom_emoji::{CustomEmoji, CustomEmojiForm},
+    custom_emoji_keyword::{CustomEmojiKeyword, CustomEmojiKeywordForm},
+  },
+  utils::{get_conn, DbPool},
+};
+
+impl CustomEmojiView {
+  pub async fn get(pool: &DbPool, emoji_id: i32) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let emoji = custom_emoji
+      .find(emoji_id)
+      .first::<CustomEmoji>(conn)
+      .await?;
+    let keywords = CustomEmojiKeyword::belonging_to(&emoji)
+      .load::<CustomEmojiKeyword>(conn)
+      .await?;
+
+    let view = CustomEmojiView {
+      custom_emoji: emoji,
+      keywords: Some(keywords),
+    };
+
+    Ok(view)
+  }
+
+  pub async fn get_all(pool: &DbPool, for_local_site_id: LocalSiteId) -> Result<Vec<Self>, Error> {
```

Comment:
Same here, there are dozens of other examples in this folder, and none of them require multiple queries.